pkg/admission/pathannotation/pathannotation_admission_test.go

```diff
@@ -0,0 +1,313 @@
+			admissionResource: apisv1alpha1.SchemeGroupVersion.WithResource("apiresourceschemas"),
+			admissionVerb:     admission.Create,
+			admissionObject:   &apisv1alpha1.APIResourceSchema{},
+			getLogicalCluster: wellKnownGetLogicalCluster,
+		},
+		{
+			name:              "an APIExport with incorrect path annotation is NOT admitted",
+			admissionVerb:     admission.Create,
+			admissionResource: apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:   &apisv1alpha1.APIExport{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{core.LogicalClusterPathAnnotationKey: "universe:milky-way"}}},
+			admissionContext:  wellKnownAdmissionContext,
+			getLogicalCluster: wellKnownGetLogicalCluster,
+			expectError:       true,
+		},
+		{
+			name:              "an APIExport without the path annotation is NOT admitted",
+			admissionVerb:     admission.Create,
+			admissionResource: apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:   &apisv1alpha1.APIExport{},
+			admissionContext:  wellKnownAdmissionContext,
+			getLogicalCluster: wellKnownGetLogicalCluster,
+			expectError:       true,
+		},
+		{
+			name:              "happy path: an APIExport with the path annotation is admitted",
+			admissionVerb:     admission.Create,
+			admissionResource: apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:   &apisv1alpha1.APIExport{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{core.LogicalClusterPathAnnotationKey: "root:foo"}}},
+			admissionContext:  wellKnownAdmissionContext,
+			getLogicalCluster: wellKnownGetLogicalCluster,
+		},
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.name, func(t *testing.T) {
+			target := &pathAnnotationPlugin{getLogicalCluster: scenario.getLogicalCluster}
+			attr := admission.NewAttributesRecord(
+				scenario.admissionObject,
+				nil,
+				schema.GroupVersionKind{},
+				"",
+				"",
+				scenario.admissionResource,
+				"",
+				scenario.admissionVerb,
+				scenario.admissionOptions,
+				false,
+				nil,
+			)
+
+			err := target.Validate(scenario.admissionContext, attr, nil)
+
+			if scenario.expectError && err == nil {
+				t.Errorf("expected to get an error")
+			}
+			if !scenario.expectError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+		})
+	}
+}
+
+func wellKnownGetLogicalCluster(clusterName logicalcluster.Name, name string) (*corev1alpha1.LogicalCluster, error) {
+	if clusterName != "test-cluster" {
+		return nil, fmt.Errorf("unexpected clusterName = %q, expected = \"test-cluster\"", clusterName)
+	}
+	if name != corev1alpha1.LogicalClusterName {
+		return nil, fmt.Errorf("unexpected name = %q, expected = %q", clusterName, corev1alpha1.LogicalClusterName)
+	}
+	return &corev1alpha1.LogicalCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: corev1alpha1.LogicalClusterName,
+			Annotations: map[string]string{
+				core.LogicalClusterPathAnnotationKey: "root:foo",
```

Comment:
Why is this `root` when the `clusterName` passed in is `test-cluster`?

Reply:
change the `clusterName` to be `foo`